File: src/test/java/j/examples/rules/condition/AssertionUtils.java
package j.examples.rules.condition;

import java.util.function.Supplier;

import static org.assertj.core.api.Assertions.assertThat;

public class AssertionUtils {
    private AssertionUtils() {}

    static void assertValid(Supplier<Boolean> supplier) {
        assertThat(supplier.get()).isTrue();
    }

    static void assertNotValid(Supplier<Boolean> supplier) {
        assertThat(supplier.get()).isFalse();
    }
}

File: src/test/kotlin/k/examples/rules/condition/AssertionUtils.kt
package k.examples.rules.condition

import org.assertj.core.api.Assertions

object AssertionUtils {

    public fun assertValid(lambda: () -> Boolean) {
        Assertions.assertThat(lambda.invoke()).isTrue()
    }

    public fun assertNotValid(lambda: () -> Boolean) {
        Assertions.assertThat(lambda.invoke()).isFalse()
    }
}